### ISSUE TITLE
Question on how to solve HI2.exe

### ISSUE THREAD

**Issue by sebastiendamaye** [1amtom/writeups#1]
Hi,

Thank you for your solution to the Sharif CTF 2016 HI2.exe challenge. It's very nice!

I'm trying to find how to set breakpoints to hook the functions you mentionned and modify the returned values. I have done it for the first function (GetSystemInfo) and have explained how I did here: https://www.aldeid.com/wiki/SharifCTF-2016/hi#Hooking_GetSystemInfo. However, when I try to reach the next breakpoints (GlobalMemoryStatusEx, GetVersion), the program never reaches them and fails with the following message: "A debugger has been found running on your system". I'm wondering how you configured your environment (OllyDbg, StrongOD) because I guess I have missed something (how to hide the fact that it's being debugged?).

Thank you in advance for your help.

**Comment by 1amtom** [comment]
Hi,

Thanks for your feedback.
Have you tried selecting also _HidePEB_, _KernelMode_ and _SkipSomeExceptions_ in StrongOD's options?

I think it should solve your problem.

**Comment by sebastiendamaye** [comment]
Hi 1amtom,
Thank you so much for your quick reply. I have setup StrongOD as recommended but it now crashes before reaching the 1st breakpoint (GetSystemInfo). It opens the executable, I run F9 once, define my 1st BP to GetSystemInfo, but immediately when I then press F9, it crashes with the message "HI2.exe stopped working". (I am using a dedicated clean OllyDbg 1.10 with only StrongOD plugin for this purpose). Maybe other OllyDbg debugging options to set up?

**Comment by sebastiendamaye** [comment]
Removing `SkipSomeExceptions` option (`HidePEB`, `KernelMode` enabled) enables me to reach the 1st BP (`GetSystemInfo`), but it fails reaching the 2nd one (`GlobalMemoryStatusEx`). Debugger detected before :(

**Comment by 1amtom** [comment]
And if you don't set up any breakpoint, and just want to run an app under debugger, does it still detect the debugger?

**Comment by sebastiendamaye** [comment]
No, indeed, when no BP is set, OllyDbg seems to be stealth (no message related to debugger detected). Though, the executable (hi2.exe) crashes.

**Comment by 1amtom** [comment]
Try deleting _*.udd_ files. Have you run ollydbg as an admin? Then try to turn off/on _KernelMode_ settings, restart debugger and see what happens.

**Comment by sebastiendamaye** [comment]
Done but no real change.... Debugger still detected, whatever the status of the `KernelMode` setting in StrongOD. Maybe something with versions? I'm running Windows 7 64bit inside VMWare workstation with OllyDbg 1.10 and StrongOD 0.4.8.892. What's your environment?
